config/initializers/diaspora_federation.rb

```diff
@@ -6,8 +6,8 @@
   config.certificate_authorities = AppConfig.environment.certificate_authorities.get
 
   config.define_callbacks do
-    on :fetch_person_for_webfinger do |handle|
-      person = Person.find_local_by_diaspora_handle(handle)
+    on :fetch_person_for_webfinger do |diaspora_id|
+      person = Person.where(diaspora_handle: diaspora_id, closed_account: false).where.not(owner: nil).take
```

Comment:
Why `take` over `first`? It's unlikely for this to ever match more than one record, but if for some reason it does, `first` would at least give a deterministic result, due ordering by PK.